nucypher/cli/actions/confirm.py

```diff
@@ -32,45 +35,77 @@
     RESTAKING_LOCK_AGREEMENT,
     WINDING_DOWN_AGREEMENT
 )
+from nucypher.config.node import CharacterConfiguration
 
 
-def confirm_deployment(emitter, deployer_interface) -> bool:
+def confirm_deployment(emitter: StdoutEmitter, deployer_interface: BlockchainDeployerInterface) -> bool:
+    """
+    Interactively confirm deployment by asking the user to type the ALL CAPS name of
+    the network they are deploying to or 'DEPLOY' if the network if not a known public chain.
+
+    Aborts if the confirmation word is incorrect.
+    """
     if deployer_interface.client.chain_name == UNKNOWN_DEVELOPMENT_CHAIN_ID or deployer_interface.client.is_local:
         expected_chain_name = 'DEPLOY'
     else:
         expected_chain_name = deployer_interface.client.chain_name
-    if click.prompt(f"Type '{expected_chain_name}' to continue") != expected_chain_name:
+    if click.prompt(f"Type '{expected_chain_name.upper()}' to continue") != expected_chain_name.upper():
         emitter.echo(ABORT_DEPLOYMENT, color='red', bold=True)
         raise click.Abort()
     return True
 
 
-def confirm_enable_restaking_lock(emitter, staking_address: str, release_period: int) -> bool:
+def confirm_enable_restaking_lock(emitter: StdoutEmitter, staking_address: str, release_period: int) -> bool:
+    """Interactively confirm enabling of the staking lock with user agreements."""
     emitter.message(RESTAKING_LOCK_AGREEMENT.format(staking_address=staking_address, release_period=release_period))
     click.confirm(CONFIRM_RESTAKING_LOCK.format(staking_address=staking_address, release_period=release_period), abort=True)
     return True
 
 
-def confirm_enable_restaking(emitter, staking_address: str) -> bool:
+def confirm_enable_restaking(emitter: StdoutEmitter, staking_address: str) -> bool:
+    """Interactively confirm enabling of the restaking with user agreements."""
     emitter.message(RESTAKING_AGREEMENT.format(staking_address=staking_address))
     click.confirm(CONFIRM_ENABLE_RESTAKING.format(staking_address=staking_address), abort=True)
     return True
 
 
-def confirm_enable_winding_down(emitter, staking_address: str) -> bool:
+def confirm_enable_winding_down(emitter: StdoutEmitter, staking_address: str) -> bool:
+    """Interactively confirm enabling of winding down with user agreements."""
     emitter.message(WINDING_DOWN_AGREEMENT)
     click.confirm(CONFIRM_ENABLE_WINDING_DOWN.format(staking_address=staking_address), abort=True)
     return True
 
 
-def confirm_staged_stake(staker_address, value, lock_periods) -> None:
-    click.confirm(CONFIRM_STAGED_STAKE.format(value=value,
+def confirm_staged_stake(staker_address: str, value: NU, lock_periods: int) -> bool:
+    """Interactively confirm a new stake reviewing all staged stake details."""
+    click.confirm(CONFIRM_STAGED_STAKE.format(nunits=str(value.to_nunits()),
+                                              tokens=str(value.to_tokens()),
```

Comment:
```suggestion
                                              tokens=value,
```
Otherwise, it would print the token amount without denomination. Also, `str()` is superfluous.

Reply:
:heavy_check_mark:  - Personally, I like the explicit `str`.